core/orchestrator.go

```diff
@@ -79,6 +80,40 @@ func (orch *orchestrator) TranscoderResults(tcId int64, res *RemoteTranscoderRes
 	orch.node.transcoderResults(tcId, res)
 }
 
+func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID ManifestID) error {
+	ticket := &pm.Ticket{
+		Recipient:         ethcommon.BytesToAddress(payment.Ticket.Recipient),
+		Sender:            ethcommon.BytesToAddress(payment.Ticket.Sender),
+		FaceValue:         new(big.Int).SetBytes(payment.Ticket.FaceValue),
+		WinProb:           new(big.Int).SetBytes(payment.Ticket.WinProb),
+		SenderNonce:       payment.Ticket.SenderNonce,
+		RecipientRandHash: ethcommon.BytesToHash(payment.Ticket.RecipientRandHash),
+	}
+	seed := new(big.Int).SetBytes(payment.Seed)
+
+	sessionID, won, err := orch.node.Recipient.ReceiveTicket(ticket, payment.Sig, seed)
+	if err != nil {
+		return errors.Wrapf(err, "error receiving ticket for payment %v for manifest %v", payment, manifestID)
+	}
+
+	if won {
+		cachePMSessionID(orch.node, manifestID, sessionID)
+	}
+
+	return nil
+}
+
+func (orch *orchestrator) TicketParams(sender ethcommon.Address) *net.TicketParams {
```

Comment:
Echoing my comment for `ProcessPayment` - I think we'll get a nil pointer exception on L107 if the node is running in off-chain mode since `orch.node.Recipient` will be nil. The rest of the orchestrator methods that depend on the presence of a ETH client first check that both the node and the ETH client are not nil. Perhaps we can add a similar check here:

```
if orch.node == nil || orch.node.Recipient == nil {
    return nil
}
```

Reply:
Thank you, fixed!